src/clock_control/watchdog.rs

```diff
@@ -0,0 +1,227 @@
+//! RTC Watchdog implementation
+//!
+//! # TODO:
+//! - Add convenience methods for configuration
+//! - Consider add default configuration for start with time only
+
+use crate::prelude::*;
+use embedded_hal::watchdog::{Watchdog, WatchdogDisable, WatchdogEnable};
+use esp32::generic::Variant::Val;
+use esp32::rtccntl::wdtconfig0::*;
+use esp32::RTCCNTL;
+
+pub type WatchdogAction = WDT_STG0_A;
+pub type WatchDogResetDuration = WDT_CPU_RESET_LENGTH_A;
+
+const WATCHDOG_UNBLOCK_KEY: u32 = 0x50D83AA1;
+
+const WATCHDOG_BLOCK_VALUE: u32 = 0x89ABCDEF;
+
+pub struct WatchDog {
+    clock_control_config: super::ClockControlConfig,
+}
+
+/// Watchdog configuration
+#[derive(Debug)]
+pub struct WatchdogConfig<
+    T1: Into<MicroSeconds>,
+    T2: Into<MicroSeconds>,
+    T3: Into<MicroSeconds>,
+    T4: Into<MicroSeconds>,
+> {
+    pub period1: T1,
+    pub action1: WatchdogAction,
+    pub period2: T2,
+    pub action2: WatchdogAction,
+    pub period3: T3,
+    pub action3: WatchdogAction,
+    pub period4: T4,
+    pub action4: WatchdogAction,
+    pub cpu_reset_duration: WatchDogResetDuration,
+    pub sys_reset_duration: WatchDogResetDuration,
+    pub pause_in_sleep: bool,
+    pub reset_cpu: [bool; 2],
```

Comment:
```suggestion
    pub reset_cpu: (bool, bool),
```

This prevents from badly indexing it `config.reset_cpu[3]` for example, and catches the problem at compile time. Also, what about giving documentation for it?